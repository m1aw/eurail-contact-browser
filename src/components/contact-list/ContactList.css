.ContactList {
    background: #EEE;
    max-width: 960px;
    margin: auto;
}

.ContactList-list {
    display: flex;
    flex-wrap: wrap;
    padding: 20px;
    border: #CCC 1px solid;
    border-top: none;
    margin: 0;
    position: relative;
}

ul.ContactList-list > * {
    flex: 50%;
    width: 50%;
    flex-grow: 0;
}

.ContactList-tabs-row {
    background: #fff;
    display: flex;
    flex-direction: row;
}

.ContactList-tabs-row > * {
    flex: 1 0;
}

.ContactList-TabButton {
    background: #EEE;
    border: #CCC 1px solid;
    padding: 5px;
    border-right: none;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    cursor: pointer;
}

.ContactList-TabButton:disabled {
    color: #aaa;
    cursor: not-allowed;
}

.ContactList-TabButton.active {
    border-bottom: none;
    box-shadow: inset 0 5px 5px -5px rgba(0,0,0,0.5);
    cursor: default;
}

.ContactList-TabButton sub {
    font-size: 0.4rem;
}

.ContactList-TabButton:last-child {
    border-right:  #CCC 1px solid;
}


@media only screen and (max-width: 768px) {
    ul.ContactList-list > * {
        flex: 100%;
        width: 100%;
    }

    .ContactList-tabs-row {
        overflow-x: scroll;
    }

}
