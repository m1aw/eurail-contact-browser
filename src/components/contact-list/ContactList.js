import React, {useState, useEffect} from 'react'
import './ContactList.css'
import {HttpService} from "../../services/http";
import {ContactItem} from "../contact-item/ContactItem";

import config from "./config";

const http = new HttpService();
export const ExpandedContactContext = React.createContext();

const ContactListTabButton = ({tab, count, handleClick, active}) => {
    const classNames = 'ContactList-TabButton' + (active ? ' active': '')
    console.log(classNames)
    return (
        <button onClick={handleClick}
                className={classNames}
                disabled={count < 1}>
            {tab} <sub>{count}</sub>
        </button>
    )
}

export const ContactList = () => {
    const [tabs, setTabs] = useState()
    const [selectedTab, setSelectedTab] = useState(config.tabs[0])
    const [selectedContact, setSelectedContact] = useState()

    useEffect(() => {
        const splitIntoTabs = (contacts) => {
            const tabs = {}
            config.tabs.forEach(tabKey => tabs[tabKey] = []);
            contacts.forEach((contact) => {
                const tab = tabs[contact.name.last.substring(0, 1).toLocaleLowerCase()]
                tab && tab.push(contact)
            })
            return tabs
        }

        http.get(config.userUrl, {results: config.numberCards})
            .then(data => data.results)
            .then(data => splitIntoTabs(data))
            .then(data => setTabs(data))
    }, [])

    return (
        <ExpandedContactContext.Provider value={[selectedContact, setSelectedContact]}>
            <div>
                <h1>{config.title}</h1>
                <div className={'ContactList'}>
                    <div className={'ContactList-tabs-row'}>
                    {tabs && Object.keys(tabs).map((key, i) =>
                        <ContactListTabButton
                            key={key} tab={key}
                            count={tabs[key].length}
                            handleClick={() => setSelectedTab(key)}
                            active={key === selectedTab}
                        />
                    )}
                    </div>
                    <ul className={'ContactList-list'}>
                        {selectedTab && tabs && tabs[selectedTab].map((contact, i) =>
                            <ContactItem key={i} contact={contact}/>
                        )}
                    </ul>
                </div>
            </div>
        </ExpandedContactContext.Provider>
    )
}
