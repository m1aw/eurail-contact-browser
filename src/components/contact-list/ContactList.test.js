import React from 'react';
import renderer from 'react-test-renderer';
import {ContactList} from "./ContactList";

const { act } = renderer;


test('ContactCard should match snapshot', () => {

    let element;
    act(() => {
        element = renderer.create(
            <ContactList/>,
        );
    });

    let tree = element.toJSON();
    expect(tree).toMatchSnapshot();

});
