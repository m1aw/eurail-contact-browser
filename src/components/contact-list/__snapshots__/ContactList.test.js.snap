// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should match snapshot 1`] = `
<div>
  <h1>
    Contact List
  </h1>
  <div
    className="ContactList"
  >
    <div
      className="ContactList-tabs-row"
    />
    <ul
      className="ContactList-list"
    />
  </div>
</div>
`;
